contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -536,7 +540,7 @@ func (s *SchedulerServer) getDriver() (driver bindings.SchedulerDriver) {
 }
 
 func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
-	if n := len(s.mesosRoles); n == 0 || n > 2 || (n == 2 && s.mesosRoles[0] != "*" && s.mesosRoles[1] != "*") {
+	if n := len(s.frameworkRoles); n == 0 || n > 2 || (n == 2 && s.frameworkRoles[0] != "*" && s.frameworkRoles[1] != "*") {
```

Comment:
Good idea! A validation function checking this invariant and optionally
failing the bootstrap w/ an error?

On Tue, Dec 8, 2015 at 4:08 PM, James DeFelice <notifications@github.com>
wrote:

> In contrib/mesos/pkg/scheduler/service/service.go
> <https://github.com/kubernetes/kubernetes/pull/18348#discussion_r46963682>
> :
>
> > @@ -536,7 +540,7 @@ func (s *SchedulerServer) getDriver() (driver bindings.SchedulerDriver) {
> >  }
> >
> >  func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
> > -	if n := len(s.mesosRoles); n == 0 || n > 2 || (n == 2 && s.mesosRoles[0] != "*" && s.mesosRoles[1] != "*") {
> > +	if n := len(s.frameworkRoles); n == 0 || n > 2 || (n == 2 && s.frameworkRoles[0] != "*" && s.frameworkRoles[1] != "*") {
>
> should we also check that defaultPodRoles is a subset of frameworkRoles?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18348/files#r46963682>.
>